Click wrap Component + styles.

biocom opened the pull request:
Adds a Component for the click wrapper. Contains simple markup and styles:
- Still needs the actual logic
- Optimized for mobile/desktop
- Currently imported in the <Footer/> component. Modify if required.

Notes:
- Further design polishing will be addressed separately.

<img width="1222" alt="Screen Shot 2021-04-20 at 18 02 42" src="https://user-images.githubusercontent.com/31534717/115428694-f04acb80-a202-11eb-9212-07db74012696.png">


<img width="419" alt="Screen Shot 2021-04-20 at 18 02 55" src="https://user-images.githubusercontent.com/31534717/115428682-ed4fdb00-a202-11eb-87e4-6b72aa97e8ff.png">

anxolin reviewed (approved):
Great!

W3stside commented:
Nice

I think that the mobile view should have left aligned checkbox and text:
![Screenshot from 2021-04-21 08-31-06](https://user-images.githubusercontent.com/21335563/115514422-fb8e0d80-a27b-11eb-985e-82bfba4bdc58.png)

biocom commented:
> Nice
> 
> I think that the mobile view should have left aligned checkbox and text:
> ![Screenshot from 2021-04-21 08-31-06](https://user-images.githubusercontent.com/21335563/115514422-fb8e0d80-a27b-11eb-985e-82bfba4bdc58.png)

Have some ideas to revise it completely, but for now the click wrap will not be implemented is my understanding.

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+  min-height: 100px;
+  padding: 16px;
+  box-shadow: rgb(0 0 0) 0 -4px 0 0, rgb(0 0 0 / 16%) 0 -9px 0 0;
+  box-sizing: border-box;
```
do you need this? we have 
```
* {
  box-sizing: border-box;
}
```
in the fixed global styles

biocom replied:
Second nature to add it :) - see it in the global styles now. I will enhance it further with:
> *, *:after, *:before { box-sizing:border-box; }

In our global styles and test it a bit.

W3stside replied:
don't do it here

biocom replied:
Do it where? I added it to `src/custom/theme/baseTheme.tsx`

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+  padding: 16px;
+  box-shadow: rgb(0 0 0) 0 -4px 0 0, rgb(0 0 0 / 16%) 0 -9px 0 0;
+  box-sizing: border-box;
+  display: flex;
```
`display: flex;` twice

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+    width: 100%;
+
+    ${({ theme }) => theme.mediaWidth.upToMedium`
+      text-align: center;
```
i think this should remain `text-align: left;` as above, with `margin: 0 auto`, and instead in the `mediaWidth` this becomes `width: some amount` so:
```css
> div > p {
    text-align: left;
    margin: 0 auto;
    width: 100%;
    ${({ theme }) => theme.mediaWidth.upToMedium`
      width: 150px; // <-- guessing on pxs here but u get the point
    ...
```

biocom replied:
Let's park this for now (click wrap will be disabled).

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+    margin: 12px 0;
+    align-items: center;
+    flex-flow: column wrap;
+    align-items: center;
```
align-items twice

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+    flex-flow: column wrap;
+    align-items: center;
+    justify-content: flex-start;
+    flex-flow: column wrap;
```
twice

W3stside commented on src/custom/components/ClickWrap/index.tsx:
```diff
@@ -0,0 +1,131 @@
+      <div>
+        <p>
+          We use cookies to provide you with the best experience and to help improve our website and application. Please
+          read our <NavLink to={'/cookie-policy'}>Cookie Policy</NavLink> for more information. By clicking &apos;Accept
```
don't need `{}` when passing string props

biocom replied:
👍🏼 was copied from the HeaderMod as a reference FYI.

W3stside replied:
/shrug

W3stside commented on src/custom/components/Footer/index.tsx:
```diff
@@ -12,18 +13,18 @@ const Wrapper = styled.div`
+  margin: auto 96px 0 32px;
   width: 100%;
 
   ${({ theme }) => theme.mediaWidth.upToMedium`
```
there's no media query here so remove this right?

W3stside replied:
actually you should display none the `FooterVersion` and children

biocom replied:
Question: What if we keep showing it, and allow the user to scroll down and see those contents?

Otherwise, in case we'll use the click wrap in the future, we'll need to load the Clickwrap probably higher up (App container).

W3stside replied:
just add the FooterVersion and children in a wrapper that is hidden on media but keep the clickwrap visible

biocom replied:
Ah that's right, added now.

W3stside commented on src/custom/theme/baseTheme.tsx:
```diff
@@ -97,6 +97,15 @@ export function themeVariables(darkMode: boolean, colorsTheme: Colors) {
+      fontWeight: '500',
+      border: `0`,
+      borderRadius: '9px',
+      boxShadow: `none`
```
why backticks here and not normal single quotes?

W3stside replied:
also above line `105`

biocom replied:
Will be fixed (also some others).

W3stside commented on src/custom/theme/styled.d.ts:
```diff
@@ -80,6 +80,14 @@ declare module 'styled-components' {
       borderRadius?: string
       boxShadow?: string
     }
+    buttonSecondary: {
```
seconded :point_up:

biocom replied:
In my view the 'Wrap ETH' was a separate special action.

Ideally for a direct Primary <-> Secondary button relation, we'd have one use the primary (orange) color and the other one blank. But for lack of a design system this might not be clear why for wrap ETH a different color was used.

Let's address this when I try to create some mini design system so we have some overview.

biocom commented:
Addressed some comments and modified the style of the clickwrap heavily:
- On desktop shows a small container, so it doesn't interfere too much with other elements.
- On tablet/desktop I think it's better to focus on the click wrap, given it's a one time thing, and otherwise stands in the way for interacting with the rest of the UI.
- Now not using ButtonSecondary anymore, as there was some conflict with the wallet address button (right corner). Now using ButtonOutlined for this in the click wrap.

<img width="1256" alt="Screen Shot 2021-04-21 at 11 55 27" src="https://user-images.githubusercontent.com/31534717/115535092-baf0bd00-a298-11eb-9f07-b4366704bbb1.png">

<img width="889" alt="Screen Shot 2021-04-21 at 11 55 02" src="https://user-images.githubusercontent.com/31534717/115535110-bfb57100-a298-11eb-9594-3a3c2efcac28.png">
<img width="413" alt="Screen Shot 2021-04-21 at 11 54 47" src="https://user-images.githubusercontent.com/31534717/115535114-c04e0780-a298-11eb-9106-cb431ed24bbf.png">

W3stside commented:
nice like the changes. make sure to mark convo as resolved if u feel it is so and if u made the change :+1:

anxolin reviewed (approved):
Awesome!

We will park the clickwrap for now until legal says otherwise. 
Nice we have it almost ready. 


Do you think we can hide the component before merging it?

biocom commented:
> Awesome!
> 
> We will park the clickwrap for now until legal says otherwise.
> Nice we have it almost ready.
> 
> Do you think we can hide the component before merging it?

yes disabled now
